components/count-badge/count-badge-icon.js

```diff
@@ -79,18 +91,31 @@ class CountBadgeIcon extends CountBadgeMixin(LitElement) {
 	}
 
 	render() {
-		const numberPadding = this.size === 'small' ? '0.55rem' : '0.7rem';
+		const numberPadding = this.size === 'small' ? '-0.65rem' : '-0.7rem';
 		const numberStyles = {
 			border: '2px solid white',
 			position: 'relative',
 			left: this.dir === 'rtl' ? 0 : 'var(--d2l-count-badge-icon-padding)',
 			right: this.dir === 'rtl' ? 'var(--d2l-count-badge-icon-padding)' : 0,
 			top: numberPadding,
-			visibility: this.skeleton ? 'hidden' : undefined
+			visibility: this.skeleton ? 'hidden' : undefined,
+			display: 'inline-block'
 		};
+
+		let horizontalPadding = '0.5rem';
+		const excessDigits = this.getNumberString().length - 4;
+		// large number strings could overflow onto the next badge,
+		// update padding in this case
+		if (excessDigits >= 0) {
+			const extraRem = excessDigits / 2;
+			horizontalPadding = `calc(0.5rem + ${extraRem}rem)`;
+		}
+		this.style.setProperty('--d2l-count-badge-icon-horizontal-padding', `${horizontalPadding}`);
 		const tabbable = (this.tabStop || this.hasTooltip) && !this.skeleton;
 		const innerHtml = html`
-			${this.renderCount(numberStyles)}
+			<div class="d2l-count-badge-number-wrapper">
+				${this.renderCount(numberStyles)}
```

Comment:
To get the overflow to work, I had to add a position: absolute wrapper to the count bubble so it would not be counted in the host's area.

Reply:
Instead of this extra `<div>`, could the absolute positioning not have been applied to `.d2l-count-badge-number`?